lib/libxdp/libxdp.c

```diff
@@ -1712,6 +1721,9 @@ struct xdp_multiprog *xdp_multiprog__get_from_ifindex(int ifindex)
 	if (xinfo.attach_mode == XDP_ATTACHED_SKB) {
 		prog_id = xinfo.skb_prog_id;
 		mode = XDP_MODE_SKB;
+	} else if (xinfo.attach_mode == XDP_ATTACHED_HW) {
+		prog_id = xinfo.hw_prog_id;
+		mode = XDP_MODE_HW;
```

Comment:
A program can be attached in HW mode at the same time as a native/generic mode program is attached; we should handle that properly...

Reply:
But that would also require change of API, right? Since right now we return one multiprogram. I don't think we can just put the hw program as one of the dispatched programs, so maybe a new field in the multiprog just for offloaded programs?

Reply:
Yeah, I was thinking a new member of `struct xdp_multiprog` for the hw prog, and a new accessor like `xdp_multiprog__main_prog()`. And we can introduce a new MULTI attach_mode like the kernel has, and return that from `xdp_multiprog__attach_mode()` if there are multiple programs attached.